primitives/benches/advz.rs

```diff
@@ -4,130 +4,139 @@
 
-#![cfg(feature = "test-srs")]
-use ark_bls12_381::Bls12_381;
-use ark_bn254::Bn254;
-use ark_ec::pairing::Pairing;
-use ark_serialize::Write;
-use ark_std::rand::RngCore;
-use criterion::{criterion_group, criterion_main, BenchmarkId, Criterion, Throughput};
-use digest::{crypto_common::generic_array::ArrayLength, Digest, DynDigest, OutputSizeUser};
-use jf_primitives::{
-    pcs::{checked_fft_size, prelude::UnivariateKzgPCS, PolynomialCommitmentScheme},
-    vid::{advz::Advz, VidScheme},
-};
-use sha2::Sha256;
+#[cfg(not(feature = "test-srs"))]
+fn main() {
+    panic!("need `test-srs` feature to run this benchmark");
+}
 
-const KB: usize = 1 << 10;
-const MB: usize = KB << 10;
+#[cfg(feature = "test-srs")]
+criterion::criterion_main!(feature_gated::benches);
 
-fn advz<E, H>(c: &mut Criterion, pairing_name: &str)
-where
-    E: Pairing,
-    // TODO(Gus) clean up nasty trait bounds upstream
-    H: Digest + DynDigest + Default + Clone + Write,
-    <<H as OutputSizeUser>::OutputSize as ArrayLength<u8>>::ArrayType: Copy,
-{
-    // play with these items
-    const RATE: usize = 4; // ratio of num_storage_nodes : polynomial_degree
-    let storage_node_counts = [600, 700, 800, 900, 1000];
-    let payload_byte_lens = [1 * MB];
+#[cfg(feature = "test-srs")]
+mod feature_gated {
+    use ark_bls12_381::Bls12_381;
+    use ark_bn254::Bn254;
+    use ark_ec::pairing::Pairing;
+    use ark_serialize::Write;
+    use ark_std::rand::RngCore;
+    use criterion::{BenchmarkId, Criterion, Throughput};
+    use digest::{crypto_common::generic_array::ArrayLength, Digest, DynDigest, OutputSizeUser};
+    use jf_primitives::{
+        pcs::{checked_fft_size, prelude::UnivariateKzgPCS, PolynomialCommitmentScheme},
+        vid::{advz::Advz, VidScheme},
+    };
+    use sha2::Sha256;
 
-    // more items as a function of the above
-    let poly_degrees_iter = storage_node_counts.iter().map(|c| c / RATE);
-    let supported_degree = poly_degrees_iter.clone().max().unwrap();
-    let vid_sizes_iter = poly_degrees_iter.zip(storage_node_counts);
-    let mut rng = jf_utils::test_rng();
-    let srs = UnivariateKzgPCS::<E>::gen_srs_for_testing(
-        &mut rng,
-        checked_fft_size(supported_degree).unwrap(),
-    )
-    .unwrap();
+    const KB: usize = 1 << 10;
+    const MB: usize = KB << 10;
 
-    // run all benches for each payload_byte_lens
-    for len in payload_byte_lens {
-        // random payload data
-        let mut payload_bytes = vec![0u8; len];
-        rng.fill_bytes(&mut payload_bytes);
+    fn advz<E, H>(c: &mut Criterion, pairing_name: &str)
+    where
+        E: Pairing,
+        // TODO(Gus) clean up nasty trait bounds upstream
+        H: Digest + DynDigest + Default + Clone + Write,
+        <<H as OutputSizeUser>::OutputSize as ArrayLength<u8>>::ArrayType: Copy,
+    {
+        // play with these items
+        const RATE: usize = 4; // ratio of num_storage_nodes : polynomial_degree
```

Comment:
Maybe just comment it as the code rate, the code length can be independent of `num_storage_nodes`.